pages/_document.js

```diff
@@ -7,6 +7,28 @@ import Document, {
 } from 'next/document'
 import { renderStaticOptimized } from 'glamor/server'
 import { fontFaces, DEFAULT_FONT_SIZE } from '@project-r/styleguide'
+import { reportError } from '../lib/errors'
+
+if (typeof window !== 'undefined') {
+  const prevErrorHandler = window.onerror
+  window.onerror = (...args) => {
+    prevErrorHandler && prevErrorHandler(...args)
+    const [msg, url, lineNo, columnNo, error] = args
+    reportError(
+      'onerror',
+      (error && error.stack) || [msg, url, lineNo, columnNo].join('\n')
+    )
+  }
+  const prevRejectionHandler = window.onunhandledrejection
+  window.onunhandledrejection = (...args) => {
+    prevRejectionHandler && prevRejectionHandler(...args)
+    const [event] = args
+    reportError(
+      'onunhandledrejection',
+      (event.reason && event.reason.stack) || event.reason
+    )
+  }
+}
```

Comment:
`pages/_document.js` only runs on the server as far as I know. We might need to move this to `_app.js`. I'll verify on Monday.

Reply:
acutally I wanted to rewrite this anyways. I'll put it back into _app.tsx 👍🏻